### PR TITLE
[IMPROVED] behavior of Conn.Close() when NATS connection is provided

### LINKED ISSUE #249: Document Conn#Close behaviour when using NatsConn(...) Option
I just spent close to a day debugging why an application I was running was crashing. Turns out it was continuously running out of file descriptors because the NATS connection was not closed [when using NatsConn(...) Option](https://github.com/nats-io/go-nats-streaming/blob/master/stan.go#L292-L308). How about documenting in https://godoc.org/github.com/nats-io/go-nats-streaming#NatsConn that using this `Option` [changes the behaviour of `Conn#Close`](https://github.com/nats-io/go-nats-streaming/blob/master/stan.go#L576)?

### PR DESCRIPTION
**Pull request by kozlovic**
Resolves #249

[ci skip]

Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

File: stan.go [hunk 1]
```diff
@@ -58,23 +58,30 @@ type Conn interface {
...
 	// Close a connection to the cluster.
+	//
 	// If there are active subscriptions at the time of the close, they are implicitly closed
 	// (not unsubscribed) by the cluster. This means that durable subscriptions are maintained.
+	//
 	// The wait on asynchronous publish calls are canceled and ErrConnectionClosed will be
-	// reported to the registered AckHandler. Note that it is possible that the cluster received
-	// and persisted successfully these messages.
+	// reported to the registered AckHandler. It is possible that the cluster received and
+	// persisted successfully these messages.
```
Comment (derekcollison) [inline stan.go]: It is possible that the cluster received and persisted these messages.

File: stan.go [hunk 2]
```diff
@@ -58,23 +58,30 @@ type Conn interface {
...
+	//
+	// If a NATS connection was provided as an option to the Connect() call, the NATS
+	// connection is not closed when this call is invoked.
```
Comment (derekcollison) [inline stan.go]: Maybe add "This connection needs to be managed by the application"

File: stan.go [hunk 3]
```diff
@@ -124,8 +131,9 @@ type Options struct {
 	// in the cluster.
 	NatsURL string
 
-	// NatsConn is a user provided low-level NATS Connection that the streaming
-	// connection will use to communicate with the cluster.
+	// NatsConn is a user provided low-level NATS connection that the streaming
+	// connection will use to communicate with the cluster. When set, closing
+	// the NATS streaming connection does not close this NATS connection.
```
Comment (derekcollison) [inline stan.go]: Maybe emphasize not -> does NOT close this NATS connection. It is the responsibility of the application to manage the lifetime of the supplied NATS connection.

**Comment by kozlovic** [comment]
@derekcollison Please have a look at the updates. If they are ok, I will squash into one before merge.

**Pull request merged**